### PR TITLE
add on demand virus scanning

### PR DESCRIPTION
**Pull request by fschade**
## Description
experimental feature, since we have virus scanning on ocis experimental we've now also have the ability to scan files "on demand". This PR adds the corresponding context action to do so.

## Related Issue
- experimental

## Motivation and Context
users should get the ability to check if files are infected. In theory this shouldnt be needed that often becaused the upload scan removes infected files immediately... but in cases where the virus is not known at the moment of uploading "on demand"  scanning helps. This can happen if the virus database is updated!

## How Has This Been Tested?
- local testing

## Screenshots (if appropriate):
Bildschirmfoto 2022-09-23 um 11 18 00

## Types of changes
- [X] New feature (non-breaking change which adds functionality)

## Checklist:
- [X] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-app-files/src/mixins/actions/virusScan.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          class: 'oc-files-actions-virus-scan-trigger'
+        }
+      ]
+    }
+  },
+  methods: {
+    async $_virus_scan_trigger({ resources }) {
+      const accessToken = this.$store.getters['runtime/auth/accessToken']
+      const httpClient = clientService.httpAuthenticated(accessToken)
+      await httpClient.get(`/remote.php/dav/postprocessing/virusscan/${resources[0].fileId}`)
```
Comment (JanAckermann) [inline packages/web-app-files/src/mixins/actions/virusScan.ts:48]: Please add the configuration.server.url from the store

File: packages/web-app-files/src/mixins/actions/virusScan.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { clientService } from 'web-pkg/src/services'
+import { isLocationSpacesActive } from '../../router'
+
+export default {
+  computed: {
+    $_virus_scan() {
```
Comment (JanAckermann) [inline packages/web-app-files/src/mixins/actions/virusScan.ts:6]: Please add _items as suffix

**Review by JanAckermann — CHANGES_REQUESTED** [review]

File: packages/web-app-files/src/mixins/actions/virusScan.ts (hunk 2 above)
Comment (fschade) [reply packages/web-app-files/src/mixins/actions/virusScan.ts]: done

File: packages/web-app-files/src/mixins/actions/virusScan.ts (hunk 1 above)
Comment (fschade) [reply packages/web-app-files/src/mixins/actions/virusScan.ts]: done

**Review by JanAckermann — APPROVED** [review]

File: packages/web-app-files/src/mixins/actions/virusScan.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          class: 'oc-files-actions-virus-scan-trigger'
+        }
+      ]
+    }
+  },
+  methods: {
+    async $_virus_scan_trigger({ resources }) {
+      const accessToken = this.$store.getters['runtime/auth/accessToken']
+      const httpClient = clientService.httpAuthenticated(accessToken)
+      const { server } = this.$store.getters.configuration
```
Comment (JammingBen) [inline packages/web-app-files/src/mixins/actions/virusScan.ts:48]: Actually we prefer using the (relatively new) `ConfigurationManager` for this. You can retrieve the URL via `configurationManager.serverUrl` then.

Comment (fschade) [reply packages/web-app-files/src/mixins/actions/virusScan.ts:48]: i keep this for now, ill add this later... release is pressing :)

**Pull request merged**